### DISCUSSION TITLE
git hub commit

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#11] [Q&A] [answered]
Hello Matt,

As per your suggestion I forked kodi_panel and proposed a (simple) pull request.  (update of the frame buffer script with direction regarding the blinking cursor)

Have I done it correctly? 
Other will comes but before making them I want to make sur that I do it correctly (I am not familiar with github) 

Other question, I have a lot of ideas for new features, and want to make a list of it.
Where should I create it so every body can see it? (and maybe work on it)

**Comment by mattblovell** [comment]
You're most recent Pull Request (adding to the codec lookup table) seemed to work fine.

Afterwards, though, I think you executed a merge from this repository *back* to your branch (rather than to the master branch of your fork).  Since a Pull Request existed on that branch, github took the update as a reason to re-open the Pull Request. 

I think the flows built into github like to enforce the practice of deleting branches once they have "served their purpose".  (There are several different schools of thought regarding how best to make use of branches.  I don't have too much of an opinion on them... yet.)

> Other question, I have a lot of ideas for new features, and want to make a list of it.
> Where should I create it so every body can see it? (and maybe work on it)

The Discussion feature is a relatively new addition to github.  Seems like it would be a nice spot for such a list, provided we get some additional readers!  I think I would prefer a To Do list in that form, leaving Issues for actual bugs that people have encountered.  Think this discussion forum will work?

Cheers!
  Matt

**Comment by mattblovell** [comment, marked as answer]
> Think this discussion forum will work?

Looks like the Discussion board has an "Ideas" label / category.  That could be a good spot to populate the list.

I don't know how many of the readers on the CE Forum thread also have github accounts.  We could potentially migrate discussion over here, if there's a critical mass of people.

**Reply by nico1080** [reply]
If we are just two people writing it does not make a lot of sense to do it github. But I think I will populate the "Ideas" label / category to see if it work as a todo list. I am not familiar with IT tools in general (I am a mechanical engineer with virtually no background in coding) but I find the github interface confusing and un-intuitive (even if it is a powerful tool).

Regarding my commit for the codec table, I put it in the "my config" branch, then cherry-pick it to my "pull request" branch and then pull requested it to your master branch. I am not sur if it is the right way to do.

In the next few days I will commit some change to "my config" branch (special treatement for some info label, PVR progress bar etc if you find some commit interesting, I will try to pull them to you using different method to see the better one to use.

**Reply by mattblovell** [reply]
If you're getting the hang of github, @nico1080, could I have you mark this question as "Answered"?

Some resources, if you haven't already found them.  I should likely read through the second one myself. :) 

- [Understanding the GitHub flow](https://guides.github.com/introduction/flow/)
- [Collaborating with issues and pull requests](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests)
